File: dsa/algorithms/array/invertingIntegers.mjs
//node --experimental-modules invertingIntegers.mjs
const invertingIntegers = (input) => {
    if(!Number(input)) 
        return 'Invalid number';

    let numbers = [...input.toString()];
    const isNegativeNumber = numbers[0] == '-';
    
    let invertedInteger = [];
    for (const number of numbers) {
        if(number != '-' && number != '+')
            invertedInteger.unshift(number);
    }

    if(isNegativeNumber)
        invertedInteger.unshift('-');

    return invertedInteger.toString();
}

console.log(invertingIntegers("12345678a9"));
console.log(invertingIntegers("+123456789"));
console.log(invertingIntegers("-123456789"));
console.log(invertingIntegers(123456789));
console.log(invertingIntegers(-987654321));

File: polygot_playground/python/dsa/array/invertingIntegers.py
def invertingIntegers(input):
    
    if (type(input) != int):
        try: input = int(input)
        except: return "Invalid number"
    
    numbers = [number for number in str(input)]
    invertedInteger = []
    for number in numbers:
        if (number != '-' and number != '+'): invertedInteger.insert(0, int(number))

    isNegativeNumber = numbers[0] == '-'
    if (isNegativeNumber): invertedInteger.insert(0, '-')

    return str(invertedInteger)

print(invertingIntegers("12345678a9"));
print(invertingIntegers("+123456789"));
print(invertingIntegers("-123456789"));
print(invertingIntegers(123456789));
print(invertingIntegers(-987654321));